cas-server-support-oauth/src/main/java/org/jasig/cas/support/oauth/web/BaseOAuthWrapperController.java

```diff
@@ -106,24 +107,28 @@ protected OAuthWebApplicationService createService(final RegisteredService regis
      * Create an authentication from a user profile.
      *
      * @param profile the given user profile
+     * @param service the registered service
      * @return the built authentication
      */
-    protected Authentication createAuthentication(final UserProfile profile) {
```

Comment:
Indeed, an access token inherits from a service ticket, but we don't have any TGT. So we cannot create an access token using the CASImpl, and I don't think we cannot validate the access token for the same reason: https://github.com/Jasig/cas/blob/master/cas-server-core/src/main/java/org/jasig/cas/CentralAuthenticationServiceImpl.java#L310

Or we need to simulate some TGT in OAuth or accommodate the existing logic we have. Maybe we should merge this one and open a new issue for that if ever it's feasible.